pkg/minikube/cruntime/containerd.go

```diff
@@ -568,13 +569,20 @@ func (r *Containerd) Restart() error {
 
 // containerdImagesPreloaded returns true if all images have been preloaded
 func containerdImagesPreloaded(runner command.Runner, images []string) bool {
-	rr, err := runner.RunCmd(exec.Command("sudo", "crictl", "images", "--output", "json"))
-	if err != nil {
+	var rr *command.RunResult
+
+	imageList := func() (err error) {
```

Comment:
hm... I am trying to think, is there a better condition to wait for instead of retrying?

Reply:
we probably don't have to wait - pls see the comment [below](https://github.com/kubernetes/minikube/pull/17658#issuecomment-1828807988)